proposals/nnnn-actors.md

```diff
@@ -359,15 +307,113 @@ extension BankAccount {
+
+```swift
+extension BankAccount: Hashable {
+  nonisolated func hash(into hasher: inout Hasher) {
+    hasher.combine(accountNumber) 
+  }  
+}
+
+extension BankAccount: CustomStringConvertible {
+  nonisolated var description: String {
+    "Bank account #\(safeAccountNumberDisplayString)"
+  }
+}
+
+let fn = BankAccount.hash(into:) // type is (BankAccount) -> (inout Hasher) -> Void
+```
+
+There are two important things to note here:
+
+1. Without the `nonisolated` modifier, the protocol conformance would be invalid, because an actor-isolated synchronous instance member cannot satisfy a synchronous protocol requirement. The compiler would produce an error such as
+```
+error: actor-isolated property "description" cannot be used to satisfy a protocol requirement
+```
+2. The `nonisolated` modifier means that the `self` parameter is not consider to reference an isolated actor. This makes references to declarations on `self` cross-actor references, so actor isolation checking will prevent them from being used:
+
+```swift
+extension BankAccount {
+  nonisolated func steal(amount: Double) {
+    balance -= amount  // error: actor-isolated property 'balance' can not be referenced on non-isolated parameter 'self'
+  }
+}  
+```
+
+Non-isolated declarations are particularly useful for adapting existing asynchronous code to actors. For example, consider an existing simple "server" protocol that uses a completion handler:
+
+```swift
+protocol Server {
+  func send<Message: MessageType>(
+    message: Message,
+    completionHandler: (Result<Message.Reply>) -> Void
+  )
+}
+```
+
+Over time, this protocol should evolve to provide `async` requirements. However, one can make an actor type conform to this protocol using a non-isolated declaration:
+
+```swift
+actor MyActorServer {
+  func send<Message: MessageType>(message: Message) async throws -> Message.Reply { ... }  // this is the "real" asynchronous implementation we want
+}
+
+extension MyActorServer : Server {
+  nonisolated func send<Message: MessageType>(
+    message: Message,
+    completionHandler: (Result<Message.Reply>) -> Void
+  ) {
+    Task.runDetached {
+      do {
+        let reply = try await send(message: message)
+        completionHandler(.success(reply))
+      } catch {
+        completionHandler(.failure(error))
+      } 
+    }
+  }
+}
+```
+
+### Cross-actor references and `Sendable` types
+
+[SE-0302][se302] introduces the `Sendable` protocol. Values of types that conform to the `Sendable` protocol are safe to share across concurrently-executing code. There are various kinds of types that work well this way: value-semantic types like `Int` and `String`, value-semantic collections of such types like `[String]` or `[Int: String]`, immutable classes, and so on.
```

Comment:
Worth calling out "classes which employ other means of synchronization" as well? Just so the list is complete?

```suggestion
[SE-0302][se302] introduces the `Sendable` protocol. Values of types that conform to the `Sendable` protocol are safe to share across concurrently-executing code. There are various kinds of types that work well this way: value-semantic types like `Int` and `String`, value-semantic collections of such types like `[String]` or `[Int: String]`, immutable classes, classes which employ other means of synchronization, and so on.
```